contracts/interchain-token-service/src/contract/execute/mod.rs

```diff
@@ -154,10 +170,72 @@ fn apply_to_hub(
             apply_to_token_deployment(storage, &source_chain, &destination_chain, deploy_token)
                 .map(Message::DeployInterchainToken)?
         }
+        Message::RegisterToken(register_token) => {
+            apply_to_register_token(storage, source_chain, register_token)
+                .map(Message::RegisterToken)?
+        }
+        Message::LinkToken(link_token) => {
+            apply_to_link_token(storage, source_chain, destination_chain, link_token)
+                .map(Message::LinkToken)?
+        }
     }
     .then(Result::Ok)
 }
 
+fn apply_to_register_token(
+    storage: &mut dyn Storage,
+    source_chain: ChainNameRaw,
+    register_token: RegisterTokenMetadata,
+) -> Result<RegisterTokenMetadata, Error> {
+    interceptors::register_custom_token(storage, source_chain, register_token.clone())?;
+    Ok(register_token)
+}
+
+fn apply_to_link_token(
+    storage: &mut dyn Storage,
+    source_chain: ChainNameRaw,
+    destination_chain: ChainNameRaw,
+    link_token: LinkToken,
+) -> Result<LinkToken, Error> {
+    let source_token = state::may_load_custom_token(
+        storage,
+        source_chain.clone(),
+        link_token.source_token_address.clone(),
+    )
+    .change_context(Error::State)?
+    .ok_or(Error::TokenNotRegistered(
+        link_token.source_token_address.clone(),
+    ))?;
+
+    deploy_token_to_source_chain(
+        storage,
+        &source_chain,
+        link_token.token_id,
+        source_token.decimals,
+    )?;
+
+    let destination_decimals = state::may_load_custom_token(
```

Comment:
What's the plan with autoscaling? Adding it in a separate PR or release, or not adding it, or want to do it differently?